frontend/static/service-worker-handler.js

```diff
@@ -1,40 +1,86 @@
-function checkWebPushNotificationSupport() {
-    if(!('serviceWorker' in navigator)) {
-        return
-    }
-    
-    if(!('PushManager' in window)) {
-        return
-    }
-
-    registerServiceWorker()
+function startWebPushNotificationFlow() {
```

Comment:
the function name is not very communicative of what it does. We could name this method to something along the lines `subscribeUserToPush`. We can try to name methods in such a way that a new person reading the code without business knowledge is able to get an idea about what is happenening

Reply:
I might need a bit more discussion for this method name, as it is doing quite a lot of work including registering service worker and sending push subscription to database. So maybe this method in itself is breaking the principle of single function doing single job, but yeah if I rename it to subscribeUserToPush then I've take out other functions calls from here and put the promise chains of returned promises separately. Quite confused in that, what are your thoughts?